modules/postgres/tests/test_postgres.py

```diff
@@ -1,21 +1,19 @@
-import sqlalchemy
-
 from testcontainers.postgres import PostgresContainer
 
 
 def test_docker_run_postgres():
-    postgres_container = PostgresContainer("postgres:9.5")
-    with postgres_container as postgres:
-        engine = sqlalchemy.create_engine(postgres.get_connection_url())
-        with engine.begin() as connection:
-            result = connection.execute(sqlalchemy.text("select version()"))
-            for row in result:
-                assert row[0].lower().startswith("postgresql 9.5")
+    # https://www.postgresql.org/support/versioning/
+    supported_versions = ["12", "13", "14", "15", "16", "latest"]
+
+    for version in supported_versions:
+        postgres_container = PostgresContainer(f"postgres:{version}")
+        with postgres_container as postgres:
+            status, msg = postgres.exec(f"pg_isready -hlocalhost -p{postgres.port} -U{postgres.username}")
 
+            assert msg.decode("utf-8").endswith("accepting connections\n")
+            assert status == 0
 
-def test_docker_run_postgres_with_driver_pg8000():
-    postgres_container = PostgresContainer("postgres:9.5", driver="pg8000")
-    with postgres_container as postgres:
-        engine = sqlalchemy.create_engine(postgres.get_connection_url())
-        with engine.begin() as connection:
-            connection.execute(sqlalchemy.text("select 1=1"))
+            status, msg = postgres.exec(f"psql -hlocalhost -p{postgres.port} -U{postgres.username} -c 'select 2*3*5*7*11*13*17 as a;' ")
+            assert "510510" in msg.decode("utf-8")
+            assert "(1 row)" in msg.decode("utf-8")
+            assert status == 0
```

Comment:
This still does not guarantee that the container is accessible form the outside, only from within the container :-( But doing that would need to install a driver in the dev dependencies or an additional tests dependency block :-(

Reply:
we can install whatever dependencies in the dev group we want (unless im missing something)

Reply:
not sure why the old one needs to be removed:

```suggestion
            assert status == 0


def test_docker_run_postgres_with_sqlalchemy():
    postgres_container = PostgresContainer("postgres:9.5")
    with postgres_container as postgres:
        engine = sqlalchemy.create_engine(postgres.get_connection_url())
        with engine.begin() as connection:
            result = connection.execute(sqlalchemy.text("select version()"))
            for row in result:
                assert row[0].lower().startswith("postgresql 9.5")
```

Reply:
> we can install whatever dependencies in the dev group we want (unless im missing something)

One reason not to: to prove that the tests can run without them installed :-)

But yes, should be fine to do so.